device.go

```diff
@@ -0,0 +1,267 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package virtcontainers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/go-ini/ini"
+)
+
+// Defining this as a variable instead of a const, to allow
+// overriding this in the tests.
+var sysIOMMUPath = "/sys/kernel/iommu_groups"
+
+var sysDevPrefix = "/sys/dev"
+
+var blockPaths = []string{"/dev/sd", "/dev/hd", "/dev/vd", "/dev/ida/"}
```

Comment:
It might be worth splitting this across multiple lines with a comment showing each type of block device supported (SCSI, IDE, virtual, Compaq Intelligent Drive Array?)